module/VuFind/src/VuFind/Controller/MyResearchController.php

```diff
@@ -845,9 +844,9 @@ public function performDeleteFavorite($id, $source)
         // Perform delete and send appropriate flash message:
         if (null !== $listID) {
             // ...Specific List
-            $table = $this->getTable('UserList');
-            $list = $table->getExisting($listID);
-            $list->removeResourcesById($user, [$id], $source);
+            $listService = $this->getListService();
+            $list = $listService->getExisting($listID);
+            $listService->removeResourcesById($user->id, $list, [$id], $source);
```

Comment:
The equivalent code in the `else` below is calling `$user->removeResourcesById`, which in turn is calling `$userResourceTable->destroyLinks`. We can't remove VuFind\Db\Table\UserResource until that code is refactored. Do you plan to do that as part of this PR, or should we address it in a follow-up?

Reply:
...and of course I just realized that I should have been looking at UserList, not UserResource, so this is all expected. Sorry for the confusion, it has been a long day. ;-)